index.bs

```diff
@@ -170,22 +170,41 @@ User Agent Hints {#http-ua-hints}
-*   <dfn for="user agent" export>platform brand</dfn> (for example: "Windows", "iOS", or "AmazingOS")
-*   <dfn for="user agent" export>platform version</dfn> (for example: "NT 6.0", "15", or "17G")
-*   <dfn for="user agent" export>platform architecture</dfn> (for example: "ARM64", or "ia32")
-*   <dfn for="user agent" export>model</dfn> (for example: "", or "Pixel 2 XL")
-*   <dfn for="user agent" export>mobileness</dfn> (for example: ?0 or ?1)
+given [=user agent=], which servers can opt-into receiving via the Client Hints infrastructure
+defined in [[I-D.ietf-httpbis-client-hints]]. The definitions below assume that each [=user agent=]
+has defined a number of properties for itself:
+
+*   <dfn for="user agent" export>brand</dfn> - The [=user agent=]'s commercial name (for example:
+      "cURL", "Edge", "The World's Best Web Browser")
+*   <dfn for="user agent" export>significant version</dfn> - The [=user agent=]'s marketing version,
+      which includes distinguishable web-exposed features (for example: "72", "3", or "12.1")
+*   <dfn for="user agent" export>full version</dfn> - The [=user agent=]'s build version (for
+      example: "72.0.3245.12", "3.14159", or "297.70E04154A")
+*   <dfn for="user agent" export>platform brand</dfn> - The [=user agent=]'s operating system's
+      commercial name. (for example: "Windows", "iOS", or "AmazingOS")
+*   <dfn for="user agent" export>platform version</dfn> - The [=user agent=]'s operating system's
+      version. (for example: "NT 6.0", "15", or "17G")
+*   <dfn for="user agent" export>platform architecture</dfn> - The [=user agent=]'s underlying CPU
+      architecture (for example: "ARM64", or "ia32")
+*   <dfn for="user agent" export>model</dfn> - The [=user agent=]'s device model (for example: "",
+      or "Pixel 2 XL")
+*   <dfn for="user agent" export>mobileness</dfn> - A boolean indicating if the [=user agent=]'s
+      device is a mobile device. (for example: ?0 or ?1)
 
 [=User agents=] SHOULD keep these strings short and to the point, but servers MUST accept arbitrary
 values for each, as they are all values constructed at the [=user agent=]'s whim.
 
+[=User agents=] SHOULD return the empty string or a fictitious value for [=platform architecture=]
+unless the user's platform is one where both the following conditions apply:
+ * Binary download of executables is likely.
+ * Different CPU architectures are likely to require different binary executable resources.
+
+[=User agents=] MUST return the empty string for [=model=] if [=mobileness=] is false or on
+platforms where the model is not typically exposed.
+
+[=User agents=] MAY return the empty string or a fictitious value for [=full version=], [=platform
+architecture=] or [=model=] in cases they consider that exposing that value increase their user's
+privacy risk beyond the user benefit of the use-cases they enable.
```

Comment:
Copying feedback from top-level comment:

The reason to expose a fictitious value would more likely be compatibility than privacy. I think it's clearer to make the MAY unconditional instead of stating a specific fuzzy reason. It's hard to see how UA authors could even make the tradeoff you described, since the spec doesn't describe either the privacy risk or the user benefit.

More generally, I don't think caring about privacy should be optional but it's better than nothing.

I'd strongly suggest dropping everything from "in cases" onward, but if you want some guidance for the MAY, I'd replace it with "for privacy, compatibility, or other reasons."

Reply:
Changed to your wording.

FWIW, I wasn't trying to convey that caring for privacy is optional, but that there's a tradeoff here between usability and privacy which should be taken into consideration.